.github/workflows/pr.yml

```diff
@@ -0,0 +1,50 @@
+name: "PR Tests"
```

Comment:
It might be worth making this super-clear that this is for CPU only - otherwise we might end up with a false sense of security that all is good when the tests pass for code that isn't being compiled.

Reply:
I intended for the unit tests workflow to be in #483 . I will remove from this PR.